vnext/Microsoft.ReactNative/Views/ViewPanel.cpp

```diff
@@ -127,12 +127,25 @@ winrt::AutomationPeer ViewPanel::OnCreateAutomationPeer() {
 
 /*static*/ void ViewPanel::SetTop(xaml::UIElement const &element, double value) {
   element.SetValue(TopProperty(), winrt::box_value<double>(value));
-  element.InvalidateArrange();
+  InvalidateForArrange(element);
 }
 
 /*static*/ void ViewPanel::SetLeft(xaml::UIElement const &element, double value) {
   element.SetValue(LeftProperty(), winrt::box_value<double>(value));
-  element.InvalidateArrange();
+  InvalidateForArrange(element);
+}
+
+void ViewPanel::InvalidateForArrange(xaml::UIElement element) {
+  // If the element's position has changed, we must invalidate the parent for arrange,
+  // as it's the parent's responsibility to arrange its children.
+  auto parent = VisualTreeHelper::GetParent(element);
+  if (parent) {
```

Comment:
nit, can fold init and decl into the if:
```suggestion
  if (auto parent = VisualTreeHelper::GetParent(element)) {
```